src/Symfony/Component/DependencyInjection/Attribute/Autowire.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * Attribute to tell a parameter how to be autowired.
+ *
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class Autowire
+{
+    public readonly string|Expression|Reference $value;
+
+    /**
+     * Use only ONE of the following.
+     *
+     * @param string|null $service    Service ID (ie "some.service")
+     * @param string|null $expression Expression (ie 'service("some.service").someMethod()')
+     * @param string|null $value      Parameter value (ie "%kernel.project_dir%/some/path")
+     */
+    public function __construct(
+        ?string $service = null,
+        ?string $expression = null,
+        ?string $value = null
```

Comment:
I think `$service` would be the most common use.

Reply:
Now, I'm wondering: what about this implementation?
```php
class Autowire
{
    public readonly string|Expression|Reference $value;

    public function __construct(
        ?string $value = null,
        ?string $service = null,
        ?string $expression = null,
    ) {
		[...]

		  if (null !== $value && str_starts_with($value, '@')) {
			    match (true) {
				        str_starts_with($value, '@@') => $value = substr($value, 1),
				        str_starts_with($value, '@=') => $expression = substr($value, 2),
				        default => $service = $value,
			      }
		  }
[...]
}
```

Reply:
Can you show some usage examples?

Reply:
Oh ok, why the `@@` check?

And also:

`#[Autowire('@=service("App\\Mail\\MailerConfiguration").getMailerMethod()')]`

But then, this would not work, correct?

`#[Autowire(SomeService::class)]`

Reply:
for `@@`, see https://github.com/symfony/symfony/blob/c1aa7b5e98d1fbc68c4594cc3f04ae2ec03c74ec/src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php#L835

> But then, this would not work, correct?

correct, this would:
`#[Autowire('@'.SomeService::class)]`

Reply:
and this also of course:
`#[Autowire(service: SomeService::class)]`

Reply:
Got it, since we're not using the Yaml loader, we'll have to duplicate the logic to parse these? What about `@!`/`@?`, should this syntax be supported as well?

Reply:
> `default => $service = $value,`

Shouldn't this remove the `@`?

Reply:
> Shouldn't this remove the @?
 yes!

> Should we handle the `@*`

no need to imho

Reply:
Ok, I'm happy with this syntax! Want me to update the PR to support?

Reply:
yes please!